abci/example/example_test.go

```diff
@@ -111,7 +111,7 @@ func dialerFunc(ctx context.Context, addr string) (net.Conn, error) {
 	return cmn.Connect(addr)
 }
 
-func testGRPCSync(t *testing.T, app *types.GRPCApplication) {
+func testGRPCSync(t *testing.T, app types.ABCIApplicationServer) {
```

Comment:
```suggestion
func testGRPCSync(t *testing.T, app types.Application) {
```

For ^ to work you will also need to change

```
func NewGRPCServer(protoAddr string, app types.ABCIApplicationServer) cmn.Service {
```

to

```
func NewGRPCServer(protoAddr string, app types.Application) cmn.Service {
```

Reply:
```
abci/server/grpc_server.go:48:49: cannot use s.app (type types.Application) as type types.ABCIApplicationServer in argument to types.RegisterABCIApplicationServer:
	types.Application does not implement types.ABCIApplicationServer (wrong type for BeginBlock method)
		have BeginBlock(types.RequestBeginBlock) types.ResponseBeginBlock
		want BeginBlock(context.Context, *types.RequestBeginBlock) (*types.ResponseBeginBlock, error)
abci/server/server.go:26:56: cannot use types.NewGRPCApplication(app) (type *types.GRPCApplication) as type types.Application in argument to NewGRPCServer:
	*types.GRPCApplication does not implement types.Application (wrong type for BeginBlock method)
		have BeginBlock(context.Context, *types.RequestBeginBlock) (*types.ResponseBeginBlock, error)
		want BeginBlock(types.RequestBeginBlock) types.ResponseBeginBlock
```

build failed with ^. maybe we can't use `types.Application` here in the end

Reply:
Reverted to `types.ABCIApplicationServer`.